src/modules/pinning.js

```diff
@@ -20,11 +21,13 @@ class Pinning {
                 this.chluIpfs.logger.warn('This node is running an IPFS client that does not implement pinning. Falling back to just retrieving the data non recursively. This will not be supported');
                 await this.chluIpfs.ipfsUtils.get(multihash);
             }
+            this.chluIpfs.events.emit('pinned', multihash);
             // broadcast successful pin
             await this.chluIpfs.room.broadcast({ type: constants.eventTypes.pinned, multihash });
         } catch (error) {
             this.chluIpfs.logger.error('IPFS Pin Error: ' + error.message);
-            return;
+            this.chluIpfs.events.emit('pin error', { multihash, error });
```

Comment:
@fazo96 are we handling these events anywhere yet?

Reply:
No, the plan was to have them show up in the UI if it makes sense to do so. Without these events, it was not possible

Reply:
OK. I see. I just wanted to understand the reason behind these events.